lightningnetwork/lnd
# brontide+lnwire+wtwire+zpay32: migrate fuzz tests

## Change Description
Fixes #6679.
Migrates fuzz tests from using go-fuzz to using Go 1.18's new standard library fuzz testing.
All fuzz tests are moved into a file called fuzz_test.go for their respective package.
~~Makefile, scripts, docs also updated.~~ Moved this to #6862 

## Steps to Test
Go to each package (brontide, lnwire, watchtower/wtwire, zpay32) and run each fuzz test individually:
Example of running the brontide Fuzz_random_actone test on its own for 30s:
`cd brontide`
`go test -fuzz Fuzz_random_actone -fuzztime 30s`

PR #6862 will allow you to test in bulk:
run `make fuzz fuzztime=1s` to run every fuzz test for 1 second
See [docs/fuzz.md](https://github.com/babinchak/lnd/blob/fuzz_make/docs/fuzz.md) for further documentation

## Pull Request Checklist
### Testing

### Code Style and Documentation

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.
